src/rdo.rs

```diff
@@ -364,39 +364,67 @@ pub fn rdo_tx_size_type<T: Pixel>(
   luma_mode: PredictionMode, ref_frames: [usize; 2], mvs: [MotionVector; 2], skip: bool
 ) -> (TxSize, TxType) {
   use crate::context::max_txsize_rect_lookup;
-  let tx_size = if !fi.tx_mode_select || !luma_mode.is_intra() {
-    max_txsize_rect_lookup[bsize as usize]
-  } else {
-    sub_tx_size_map[max_txsize_rect_lookup[bsize as usize] as usize]  // 1 level partition
-  };
-  debug_assert!(tx_size.width_log2() <= bsize.width_log2());
-  debug_assert!(tx_size.height_log2() <= bsize.height_log2());
 
-  // Luma plane transform type decision
+  let mut tx_size = max_txsize_rect_lookup[bsize as usize];
+  let mut best_tx_type = TxType::DCT_DCT;
+  let mut best_tx_size = tx_size;
+  let mut best_rd = std::f64::MAX;
   let is_inter = !luma_mode.is_intra();
-  let tx_set = get_tx_set(tx_size, is_inter, fi.use_reduced_tx_set);
+  let mut tx_set;
```

Comment:
Couldn't it be declared inside the loop (where it is assigned)?

Reply:
Done. Thanks for catching these! It also saved SIX lines!
(Somehow, the message I entered did not go through bcz I didn't hit "Comment".